Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs

```diff
@@ -0,0 +1,72 @@
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Objectives;
+using Content.Server.Terminator.Components;
+using Content.Server.Terminator.Systems;
+using Robust.Shared.Map;
+
+namespace Content.Server.GameTicking.Rules;
+
+public sealed class TerminatorRuleSystem : GameRuleSystem<TerminatorRuleComponent>
+{
+    [Dependency] private readonly TerminatorSystem _terminator = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TerminatorRuleComponent, ObjectivesTextGetInfoEvent>(OnObjectivesTextGetInfo);
+    }
+
+    private void OnObjectivesTextGetInfo(EntityUid uid, TerminatorRuleComponent comp, ref ObjectivesTextGetInfoEvent args)
+    {
+        args.Minds = comp.Minds;
+        args.AgentName = Loc.GetString("terminator-round-end-agent-name");
+    }
+
+    /// <summary>
+    /// Finds an existing rule for a target, or starts a new rule and returns it.
+    /// </summary>
+    public EntityUid GetRule(EntityUid target)
+    {
+        var query = EntityQueryEnumerator<TerminatorRuleComponent>();
+        while (query.MoveNext(out var uid, out var rule))
+        {
+            if (rule.Target == target)
+                return uid;
+        }
+
+        // start a new rule
+        GameTicker.StartGameRule("Terminator", out var ruleId);
+        var comp = Comp<TerminatorRuleComponent>(ruleId);
+        comp.Target = target;
+        return ruleId;
+    }
```

Comment:
Needs to support multiple gamerules on a target, just make each one a terminator prolly.

Reply:
then objectives wont be grouped and if theres 2 terminators that targeted same person (still extremely low odds but possible) itll say
there was 1 t-800
abc kill bob

there was 1 t-800
xyz kill bob

Reply:
so should i still do it given above will happen...

Reply:
That's what happens with events like initial infected, so it's probably fine